src/OVAL/probes/unix/linux/fwupdsecattr_probe.c

```diff
@@ -0,0 +1,361 @@
+
+	if (!dbus_message_iter_init(msg, &args)) {
+		dD("Failed to initialize iterator over received dbus message.");
+		goto cleanup;
+	}
+
+	if (dbus_message_get_type(msg) == DBUS_MESSAGE_TYPE_ERROR) {
+		dD("Receive an error exception from dBus");
+		goto cleanup;
+	}
+	
+
+	if (dbus_message_iter_get_arg_type(&args) != DBUS_TYPE_ARRAY && dbus_message_iter_get_element_type(&args) != DBUS_TYPE_DICT_ENTRY) {
+		dD("Expected array of dict_entry argument in reply. Instead received: %s.", dbus_message_type_to_string(dbus_message_iter_get_arg_type(&args)));
+		goto cleanup;
+	}
+
+	// Array of[Dict of {String, Valiant}]
+	dbus_message_iter_recurse(&args, &property_iter);
+	do {
+		DBusMessageIter array_entry, dict_entry, value_variant;
+
+		/* Process array */
+		if (dbus_message_iter_get_arg_type(&property_iter) != DBUS_TYPE_ARRAY) {
+			dD("Expected property_iter is an array but recieveing a %c", dbus_message_iter_get_arg_type(&property_iter));
+			goto cleanup;
+		}
+		dbus_message_iter_recurse(&property_iter, &array_entry);
+		
+		if (dbus_message_iter_get_arg_type(&array_entry) != DBUS_TYPE_DICT_ENTRY) {
+			dD("Expected array_entry is an dict but recieveing a %c", dbus_message_iter_get_arg_type(&array_entry));
+			goto cleanup;
+		}
+
+		do {
+			dbus_message_iter_recurse(&array_entry, &dict_entry);
+
+			if (dbus_message_iter_get_arg_type(&dict_entry) != DBUS_TYPE_STRING) {
+				dD("Expected dict_entry is an string but receiving a %s", dbus_message_type_to_string(dbus_message_iter_get_arg_type(&dict_entry)));
+			}
+
+			_DBusBasicValue value;
+			dbus_message_iter_get_basic(&dict_entry, &value);
+			property_name = oscap_strdup(value.str);
+			dD("Element key: %s", property_name);
+			
+			if (dbus_message_iter_next(&dict_entry) == false) {
+				dW("Expected another field in dict_entry.");
+				free(property_name);
+				goto cleanup;
+			}
+
+			if (dbus_message_iter_get_arg_type(&dict_entry) != DBUS_TYPE_VARIANT) {
+				dW("Expected variant as value in dict_entry. Instead received: %s.", dbus_message_type_to_string(dbus_message_iter_get_arg_type(&dict_entry)));
+				free(property_name);
+				goto cleanup;
+			}
+
+			dbus_message_iter_recurse(&dict_entry, &value_variant);
+
+			const int arg_type = dbus_message_iter_get_arg_type(&value_variant);
+
+			switch (arg_type) {
+				case DBUS_TYPE_UINT32:
+					if(strncmp(property_name, "HsiResult", strlen("HsiResult")) == 0) {
+						_DBusBasicValue hsiresult_value;
+						dbus_message_iter_get_basic(&value_variant, &hsiresult_value);
+						hsi_flags = hsiresult_value.u32;
+					}
+					break;
+				case DBUS_TYPE_STRING:
+					if(!strncmp(property_name, "AppstreamId", strlen("AppstreamId"))) {
+						appstream_name = dbus_value_to_string(&value_variant);
+						dD("Element string: %s", appstream_name);
+					}
+					break;
+			}
+			free(property_name);
+		} while (dbus_message_iter_next(&array_entry));
+		callback(appstream_name, hsi_flags);
```

Comment:
What is going to happen if `appstream_name` is still `NULL` and `hsi_flags` is `0`?

Reply:
Skip it? Since appstream_id shouldn't be NULL.